adapters/adapter.go

```diff
@@ -9,32 +9,42 @@ import (
 	"time"
 )
 
+// Adapters connect prebid-server to a demand partner. Their primary purpose is to produce bids
+// in response to Auction requests.
 type Adapter interface {
+	// A name which uniquely identifies this adapter.
+	// This must not overlap with any other adapters in prebid-server.
 	Name() string
 	FamilyName() string
 	GetUsersyncInfo() *pbs.UsersyncInfo
+	// Produce bids which should be considered, given the auction params.
```

Comment:
aha... ok. I think I can reconcile these things with careful wording.

At first I imagined something like "this makes an external server call." But then implementations which _don't_ make that call would be breaking [Liskov's Substitution Principle](https://en.wikipedia.org/wiki/Liskov_substitution_principle)--which would be limiting.

Can be descriptive without limiting if it says something like "this usually makes an external call, although the interface doesn't require it," though.